Sources/SKCore/BuildServerBuildSystem.swift

```diff
@@ -135,16 +135,22 @@ final class BuildServerHandler: LanguageServerEndpoint {
   }
 
   func handleFileOptionsChanged(_ notification: Notification<FileOptionsChangedNotification>) {
-    // TODO: add delegate method to include the changed settings directly
-    self.delegate?.fileBuildSettingsChanged([notification.params.uri])
+    let params = notification.params
+    let options = params.updatedOptions
+    let fileBuildSettings = FileBuildSettings(
+      compilerArguments: options.options, workingDirectory: options.workingDirectory)
+    self.delegate?.fileBuildSettingsChanged([params.uri: .modified(fileBuildSettings)])
   }
 }
 
 extension BuildServerBuildSystem: BuildSystem {
 
   /// Register the given file for build-system level change notifications, such as command
   /// line flag changes, dependency changes, etc.
-  public func registerForChangeNotifications(for url: LanguageServerProtocol.URL) {
+  public func registerForChangeNotifications(
```

Comment:
The build server still has a `settings` request; I think we need to make that request here to get the initial settings.

Reply:
Hmm I believe this is still necessary right?

Reply:
@rmaz what do you think? should we change the protocol or manually fetch the settings here?

Reply:
The register request should be enough to get the settings, the initial settings should be returned as soon as they are available. I don't believe we need the `textDocument/sourceKitOptions` request any more.

Reply:
Okay, left this as is, we can remove the `textDocument/sourceKitOptions` in a follow up if this is working well

Reply:
The way `FileOptionsChangedNotification` and registration is setup does not play nicely with files that are outside the build model.  Right now, if a file is not known to the build server, it's not clear if it will fail to register (in which case it will not get a notification if it is later added), or if it will register but not provide initial "nil" settings (since `FileOptionsChangedNotification` does not have a way to represent lack of settings).  For the same reason, a file that successfully registers will never have its settings removed.

In the previous model, we could handle this because the settings update would trigger re-querying the settings synchronously, which can fail.

Reply:
If the server can tell that the file is outside, it should fail registration (and we handle that OK). Otherwise it should provide initial nil settings.

Reply:
What will happen in these two cases?

1. file that starts unknown to build system is added to the build system (e.g. after you edit a config file).  Do we ever get notified of the settings? I'm guessing not since registration failed.

2. file that is known by the build system is removed from the build system. How can we be notified that this happened when the update notification has no notion of nil/removed?

Reply:
Ah, by `If the server can tell that the file is outside, it should fail registration (and we handle that OK). Otherwise it should provide initial nil settings.` I meant that the server should only fail registration if it knows that file can never be handled by the build system (e.g. for a monorepo this happens if it's in a different directory). Unless it can do that it should register and provide no settings.

re #2 hmm this we won't handle, @rmaz what do you think?